CONTRIBUTING

Dan-Burns opened the pull request:
<!--
# TODO: If you're a maintainer of this repo, open `.github/PULL_REQUEST_TEMPLATE.md`, replace <NAME> with the repo name, and remove this line.

Many thanks for contributing to Arcadia-Science/<NAME>!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).
-->

## PR checklist

- [ x] Tag the issue(s) or milestones this PR fixes (e.g. `Fixes #123, Resolves #456`).
    - Adds support for GetContacts and py-mfdca (no issue number yet).
    - Not using py-mfdca in current notebook but useful for further exploration.

- [ x] Describe the changes you've made.
- - Added `scripts/setup_extras.sh` to install GetContacts and py-mfdca into `.external/`.
    - Updated `.gitignore` to exclude `.external/`.
    - Added P00813_alignment.fasta to input/ for use with py-mfdca.
    - Updated `env.yml` to include dependencies (e.g., vmd-python).
    - Added documentation to README and SETUP.
    - Added cells in index.ipynb to generate getcontacts data and display the contact map
- [x ] Describe any tests you have conducted to confirm that your changes behave as expected.
   - make lint and make format pass
- [ x] If you've added new software dependencies, make sure that those dependencies are included in the appropriate `conda` environments.
- [ ] If you've added new functionality, make sure that the documentation is updated accordingly.
- [ ] If you encountered bugs or features that you won't address, but should be addressed eventually, create new issues for them.

Dan-Burns committed:
CONTRIBUTING

ekiefl commented on SETUP.qmd:
```diff
@@ -24,7 +24,7 @@ git clone https://github.com/{{< var pub.org >}}/{{< var pub.repo >}}.git
 :::
 
-This repository uses conda to manage the computational and build environment. If you don't have it installed (check with `conda --version`), you can find operating system-specific instructions for installing miniconda [here](https://docs.conda.io/projects/miniconda/en/latest/). After installing, run the following commands to create and activate the environment.
+This repository uses conda to manage the computational and build environment. If you don't have it installed (check with `conda --version`), you can find operating system-specific instructions for installing miniconda [here](https://docs.conda.io/projects/miniconda/en/latest/). After installing, run the following commands to create and activate the environment. If you run into issues creating the ...
```
Did you have trouble with conda?

Dan-Burns replied:
Thanks @ekiefl - yes, please push anything you like.

I did have trouble with conda but no issues with mamba.

ekiefl commented on _quarto.yml:
Awesome. I know it's a bit awkward to run `make bump-version` *before* you make any changes, so I appreciate that you read the instructions carefully. (Versioning has been smoothed out significantly in the current notebook pub template, and it's now it's fully automated by GitHub Actions).

ekiefl commented on env.yml:
```diff
@@ -11,6 +12,13 @@ dependencies:
+  - vmd-python
+  - pytest
+  - seaborn
+  - cython
```
Are these all `getcontacts` requirements? We'll need to pin these to create a reproducible environment.

Why `pytest`?

ekiefl replied:
Could you please sort out which of these deps are truly required?

Dan-Burns replied:
Pasted from the chatbot. Cut it to just vmd-python.

ekiefl commented on setup_extras.sh:
```diff
@@ -0,0 +1,33 @@
+# Add GetContacts to PATH for this session
+export PATH="$(pwd)/getcontacts:$PATH"
+echo "🔗 GetContacts added to PATH for this session."
+
+# === Clone and install py-mfdca ===
+if [ ! -d py-mfdca ]; then
+    echo "📦 Cloning py-mfdca..."
+    git clone https://github.com/utdal/py-mfdca.git
+else
+    echo "✅ py-mfdca already cloned."
+fi
+
+cd py-mfdca
+echo "🔧 Installing py-mfdca..."
+pip install .
+cd ../..
+
+echo "✅ All extra packages installed successfully."
```
Since it's not part of the paper, I don't think it should piggy back on this PR.

Dan-Burns replied:
I added a final analysis that uses it. If you don't think it adds value, I can remove the analysis and the package.

ekiefl commented on setup_extras.sh:
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -e
```
Interestingly, this was killing my shell, so I think it should be removed. I'm zsh 🤷

Dan-Burns replied:
Gone.

ekiefl commented on setup_extras.sh:
I understand the rationale for this script: getcontacts does not have a straight forward install directive, so this script remedies that. But it's unfortunate that the getcontacts `get_*` scripts only last for the session... This makes `source setup_extras.sh` an activation instruction as much as it is an install instruction.

I think we might be better off adding getcontacts as a submodule so we can manage the PATHs permanently by adding them to the pyproject.toml. They would be baked in as entry points during `pip install -e .`, making them proper CL tools, which would forego the need to modify the install instructions at all, or the need to add ephemeral modifications to PATH.

But ideally the true fix lies upstream of this repo. Rather than inheriting this nonideality, I'd rather have getcontacts be properly installable (https://github.com/getcontacts/getcontacts/issues/101). Unfortunately, it doesn't look actively maintained. I see you've contributed to it in the past. You don't have maintainer privileges do you? If not, what was your experience with the maintainer?

Dan-Burns replied:
I don't think he's active on it at all. I've had another PR sitting on it for a couple years.

The sub module route should be working now. You still have to the extra step of sourcing .bashrc - and then activating the environment a second time.

ekiefl reviewed (commented):
This is really cool @Dan-Burns, thanks for whipping this up. Looking at your addition, I can see how this provides a clearer picture of the true "contact map". I feel like the current pub almost uses the term incorrectly--it's more of an inverse distance map.

![image](https://github.com/user-attachments/assets/2b5040ab-e8d9-4c07-b227-e6bea6eddcca)

In particular, there is a lot less emphasize on the off-diagonal, which better reflects what we observe in the 150M model.

---

Regarding the code itself, it's really a shame that getcontacts isn't a proper package. I did propose an alternative installation strategy below, which involves adding getcontacts as a submodule and adding the get_ scripts as entrypoints. Do I have your permission to push these commits to your branch?

Dan-Burns committed:
Remove unnecessary dependencies from env.yml. Added additional notebook section and citation

Dan-Burns committed:
Edited text before final figure.

ekiefl commented:
Hey @Dan-Burns, thanks for taking another stab. I'm planning on joining in on Monday. It's one of the first things on my list for next week. Have a nice weekend!

Dan-Burns commented:
Thanks @ekiefl. Enjoy your weekend too!

On Fri, May 30, 2025 at 5:44 PM Evan Kiefl ***@***.***> wrote:

> *ekiefl* left a comment (Arcadia-Science/2024-paired-token-masking#12)
> <https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12#issuecomment-2923728109>
>
> Hey @Dan-Burns <https://github.com/Dan-Burns>, thanks for taking another
> stab. I'm planning on joining in on Monday. It's one of the first things on
> my list for next week. Have a nice weekend!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12#issuecomment-2923728109>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN7D4BRTZQAQOMSLEDOLXED3BDUNLAVCNFSM6AAAAAB6BXRQEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMRTG4ZDQMJQHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dan-Burns committed:
Changed to submodules and installing dependendies with make install.

Dan-Burns committed:
fix py-mfdca install in makefile

Dan-Burns committed:
pinned numpy 2.1.0 for py-mfdca

ekiefl committed:
Don't ignore gitmodules

ekiefl committed:
Unpin unsatisfiable numpy version

- vmd-python requires numpy<2

ekiefl committed:
.deps -> third_party (visibility)

ekiefl committed:
Remove py-mfdca as gitsubmodule -- install with pip

- Driveby: exclude third_party from linting/formatting

ekiefl committed:
Add getcontacts scripts as package entrypoints

ekiefl committed:
Update installation instructions

- Revert all ad hoc install instructions

ekiefl committed:
Run `make execute`

- np.bool = np.bool_ trick for numpy compatibility with py-mfdca
- Add new scripts
- Required rolling getcontacts back to before this PR:
  https://github.com/getcontacts/getcontacts/commit/da14deb263635e1e555519e5f208c61e83148eef.
  For whatever reason, the output was coming back empty.

ekiefl committed:
Don't track getcontacts outputs

- Touch up text

ekiefl commented:
Hey @Dan-Burns I made a number of changes. Here's where we're at now.

## Infra

* I removed py-mfdca as a git submodule since the repo is pip installable
* I noticed you pinned numpy > 2, but from what I can tell, vmd requires numpy < 2
* Renamed .deps to third_party and removed it from the gitignore -- these modules should be tracked
* I added the getcontacts scrips as package entrypoints. This means they're added to PATH when the user does a `pip install -e .`
* This allowed me to roll back all the bespoke install instructions, so now we're back to creating a conda environment and running `pip install -e .`
* I had to revert [this getcontacts commit](https://github.com/getcontacts/getcontacts/commit/da14deb263635e1e555519e5f208c61e83148eef) in order for get_contact_frequencies.py to spit out anything. I presume you didn't run into this, and I'm not sure why.

Given all these changes, it would be useful if you could do a fresh install and verify that `make execute` works and the plots all show up. Remember to `git submodule update --init --recursive`.

## Analysis

I noticed that the contact map heatmap `hm` is shape (360, 360), but ADA is (363, 363). Consequently, I think there's some alignment issues in Figure 8 that need to be resolved.

Your MSA energy analysis is really cool. I'm still wrapping my head around it, but I like the approach of using these more interpretable logits to try and tease out other patterns of pairwise dependency. That said, I don't see many notable reflections -- at least, it's hard for me to discern spurious correlation with something more consistent/systematic.

If you want, I would welcome a more quantitative comparison. Otherwise, I think we should either scrap the DCA analysis or soften the language and present it as a null result in an addendum. What do you think?

![image](https://github.com/user-attachments/assets/7f48f4b2-aae1-4e3a-a96a-83f7cfc53df1)

Dan-Burns committed:
removed dca contact map

Dan-Burns commented:
> * I noticed you pinned numpy > 2, but from what I can tell, vmd requires numpy < 2

I don't know what I did with my environment but I had to update numpy to get the py-mfdca part to work this morning. 


> Given all these changes, it would be useful if you could do a fresh install and verify that `make execute` works and the plots all show up. Remember to `git submodule update --init --recursive`.

I cloned the latest version, made a new environment, and installed everything. Looks like it's working. 

> I noticed that the contact map heatmap `hm` is shape (360, 360), but ADA is (363, 363). Consequently, I think there's some alignment issues in Figure 8 that need to be resolved.

I added a few lines to find the missing residues and add them in.

> Your MSA energy analysis is really cool. I'm still wrapping my head around it, but I like the approach of using these more interpretable logits to try and tease out other patterns of pairwise dependency. That said, I don't see many notable reflections -- at least, it's hard for me to discern spurious correlation with something more consistent/systematic.

It's a bit of a stretch. The trick is to hold your breath for about a minute and then squint really hard to see them. 

> If you want, I would welcome a more quantitative comparison. Otherwise, I think we should either scrap the DCA analysis or soften the language and present it as a null result in an addendum. What do you think?

I think you're right - I removed it entirely.

ekiefl commented:
Sounds good. We're getting close!

> I added a few lines to find the missing residues and add them in.

Can you push these changes? Last I see if you removed the py-mfdca dep: [c7fd80b](https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12/commits/c7fd80bf07c0d26b544c8d07f1c5622d361c3aea)

ekiefl commented:
My mistake, I see the index.ipynb has been changed. Looking now

ekiefl committed:
Remove DCA ref in ref.bib

ekiefl committed:
Remove unused alignment

ekiefl committed:
Flush out some of the text

- Run `make execute`

ekiefl commented:
@Dan-Burns, do you have an ORCID you would like associated with your name?

Dan-Burns commented:
@ekiefl yes, thanks! 0000-0003-0739-2070

ekiefl committed:
Add Daniel Burns author information

ekiefl committed:
With last

ekiefl commented:
Hey @Dan-Burns, finally got the green light on publishing the revision. Are you OK with this affiliation or would you prefer something else? https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12/commits/5a9ba56ec5bc9ed0b18b694a5ca415d742b79ecd

Dan-Burns commented:
Hi @ekiefl - Can we leave affiliation blank?

ekiefl commented:
@Dan-Burns, probably! Let's try and see if it complains.

ekiefl committed:
Run `make execute`

ekiefl commented:
@Dan-Burns, changes are live! Sorry it took so long, and congrats on your contribution 🥳 https://arcadia-science.github.io/2024-paired-token-masking/

Dan-Burns commented:
Great - Thanks!

On Wed, Jul 30, 2025 at 2:40 PM Evan Kiefl ***@***.***> wrote:

> *ekiefl* left a comment (Arcadia-Science/2024-paired-token-masking#12)
> <https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12#issuecomment-3137757098>
>
> @Dan-Burns <https://github.com/Dan-Burns>, changes are live! Sorry it
> took so long, and congrats on your contribution 🥳
> https://arcadia-science.github.io/2024-paired-token-masking/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Arcadia-Science/2024-paired-token-masking/pull/12#issuecomment-3137757098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN7D4BWZ7DYOXHORWELLWB33LEUTZAVCNFSM6AAAAAB6BXRQEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCMZXG42TOMBZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
